File: 프로그래머스/2/86971. 전력망을 둘로 나누기/전력망을 둘로 나누기.js
function search(n, wires) {
    let visited = Array.from({length : n+1}, () => 0);
    let queue = [];
    queue.push(1);
    visited[1] = 1;
    let cnt =0;
    while(queue.length !== 0) {
        const tower = queue[0];
        queue.shift();
        
        cnt +=1;
        
        for(const item of wires) {
            
            if(item[0] === tower && !visited[item[1]]) {
                queue.push(item[1]);
                visited[item[1]] = 1;
                continue;
            }
            
            if(item[1] === tower && !visited[item[0]]) {
                queue.push(item[0]);
                visited[item[0]] =1;
                continue;
            }
        }
    }
    
    return Math.abs(n-2*cnt);
}

function solution(n, wires) {
    var answer = 101;
    
    const len = wires.length;
    for(let i=0;i<len ; i++) {
        const arr = wires.filter((item, idx) => idx !== i);
        const res = search(n, arr);
        
        if(res === 0) {
            answer = 0;
            break;
        } else if(res< answer) {
            answer = res;
        }
    }
    
    
    return answer;
}

File: 프로그래머스/2/86971. 전력망을 둘로 나누기/전력망을 둘로 나누기.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int search(vector<bool> & visited,vector<vector<int>> &wires) {
    queue<int> q;
    
    int len = wires.size();
    int cnt =1;
    q.push(wires[0][0]);
    visited[wires[0][0]] = true;
    
    while(!q.empty()) {
        int tower = q.front();
        q.pop();
        
        for(int i=0; i<len; i++) {
            vector<int> wire = wires[i];
            
            if(wire[0] == tower && !visited[wire[1]]) {
                q.push(wire[1]);
                cnt +=1;
                visited[wire[1]] = true;
            }
            
            if(wire[1] == tower && !visited[wire[0]]) {
                q.push(wire[0]);
                cnt +=1;
                visited[wire[0]] = true;
            }
        }
    }
    
    return cnt;
    
}

int solution(int n, vector<vector<int>> wires) {
    int answer = n;
    
    for(int i=0; i<n-1;i++) {
        vector<vector<int>> tmp = wires;
        tmp.erase(tmp.begin() + i);
        vector<bool> visited(n+1);
        
        int res = search(visited,tmp);
        res = abs(2*res -n);
        
        if(res == 0) {
            answer = 0;
            break;
        }
        else {
            if(res < answer) answer = res;
        }
    }
    
    
    return answer;
}